args/challenges.py

```diff
@@ -17,9 +17,45 @@ def parse(parser):
                             help = "Remove character/esper rewards from: Auction House, Collapsing House, Figaro Castle Throne, Gau's Father's House, Kohlingen Inn, Narshe Weapon Shop, Sealed Gate, South Figaro Basement")
     challenges.add_argument("-pd", "--permadeath", action = "store_true",
                             help = "Life spells cannot be learned. Fenix Downs unavailable (except from starting items). Buckets/inns/tents/events do not revive characters. Phoenix casts Life 3 on party instead of Life")
+    challenges.add_argument("-rls", "--remove-learnable-spells", type = str,
+                            help = "Remove spells from learnable sources: Items, Espers, Natural Magic, and Objectives")
 
 def process(args):
-    pass
+    from constants.spells import black_magic_ids, white_magic_ids, gray_magic_ids, spell_id
+
+    # If no_ultima is on, add it to our exclude list for downstream use
+    # If permadeath is on, add it to our exclude list for downstream use
+    args.remove_learnable_spell_ids = []
+    if args.no_ultima:
```

Comment:
I know in my remove-item branch I removed the old flags as I figured the UI could just map no_ultima input to the given flag, but this is definitely the play to retain flag compatibility and ease the transition to these flags.

Reply:
Yeah, I think they'll be a great scrubbing of the flags at some point in the future in which duplicate flags will be removed (and flags that no one will ever change, like Auto-Sprint will just become on by default). I figured I'd leave the existing ones in place for now.